### DISCUSSION TITLE
Minimum JDK for opencrx 5.1

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx-documentation#28] [Q&A] [answered]
I am currently on opencrx 4.1.  I am considering an upgrade to version 5.1. 
What is the minimum jdk for version 5.1 ?

 Can it run on java 8 ? 

Due to some other constraints I can't deploy on a java 11 environment.

**Comment by wfro64** [comment]
The binaries published on Maven are built with the target platform VERSION_11. See [here](https://github.com/opencrx/opencrx/blob/master/build.properties). So openCRX requires Java 11.

There should be no problem to build openCRX for VESION_1_8.

**Reply by nyemiks** [reply]
Good news. I will like to try this out.

So if I want to build using java 8 all I need do is change the target.platform=VERSION_11

To

target.platform=VERSION_1_8 
In the build.properties file and do a rebuild ?

Is that all that needs to be done ?

**Comment by wfro64** [comment, marked as answer]
You can build openCRX for Java 8 as follows:

Set JDK 8:

```
java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-8u292-b10-0ubuntu1~20.04-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)

```
Get openCRX:

```
git clone git@github.com:opencrx/opencrx.git opencrx
cd opencrx
```

Replace `JavaVersion.VERSION_11` to `JavaVersion.VERSION_1_8` in the following files:

```
modified:   build.gradle.kts
modified:   build.properties

```

Disable option

`options.compilerArgs = listOf("--add-exports", "java.naming/com.sun.jndi.ldap=ALL-UNNAMED")`

in

`modified:   core/build.gradle.kts`

**Reply by nyemiks** [reply]
I have been able to build from source with java 8. Thanks !
